### DISCUSSION TITLE
Problem with weeks numbers ?

### DISCUSSION THREAD

**Discussion by mojimbo80** [rsuite/rsuite#3467] [Q&A]
Hi all, 
I have a question about the values of the weeks number.
When i'm not use the property "isoWeek", i don't have the 53th week in 2020 ( it's a year with 53 weeks).
Is it normal ? 
<img width="714" alt="Screenshot 2023-11-24 at 06 47 38" src="https://github.com/rsuite/rsuite/assets/41619907/53faf74f-d3f3-4ee2-948a-689407929a01">
(It might be a misunderstanding on my part)

**Comment by SevenOutman** [comment]
That could be about the definition of the first week of the year. DatePicker and DateRangePicker are using date-fns `format` function to get the week number, which by default regards the week that January 1st falls in the first week of year, while ISO regards the week January 4th falls in the first week of year. date-fns `format` has an option for customizing that behavior, but don't know why it doesn't default to Jan 4.

ISO week number wiki https://en.wikipedia.org/wiki/ISO_week_date

`format` option https://date-fns.org/v2.16.1/docs/format#arguments

Related issue in date-fns https://github.com/date-fns/date-fns/issues/1945#issuecomment-1568930180

**Reply by mojimbo80** [reply]
Thank you for these very clear explanations, i learn a lot.

The problem with this situation is that impact others years.
Exemple : 2022 ( the date-range-picker shows 53 weeks but in reality it's 52 )
<img width="714" alt="Screenshot 2023-11-25 at 08 49 53" src="https://github.com/rsuite/rsuite/assets/41619907/c66b31ce-36fc-4966-b1e5-9124b37b8f6a">

The feature showWeekNumbers (which is very usefull ) seems degraded  for the users who wants a calendar in the american weeks system (sunday => saturday)
Did you know, if it possible to fix that in future update ?

**Reply by SevenOutman** [reply]
It depends on the team what actions to take. But I think it should be technically fixable as far as I'm concerned.
